packages/worker/src/worker-options.ts

```diff
@@ -752,6 +754,8 @@ export function compileWorkerOptions(opts: WorkerOptionsWithDefaults): CompiledW
     opts.maxConcurrentWorkflowTaskExecutions = 2;
   }
 
+  const activities = new Map(Object.entries(opts.activities ?? {}).filter(([_, v]) => typeof v === 'function'));
```

Comment:
This doesn't work with `Proxy` and non-enumerable properties, let's not break compatibility.
I wouldn't modify the activities object, but you can disable non local activities.

Reply:
Fair point, though, to be exact, the behavior with `Proxy` and non-enumerable properties was already ill-defined and incorrect, so I'm not that much worried about breaking compatibility.

For example, starting a worker with `activities` defined like this would make it a workflow-only worker. Same if `activities` only contain non-enumerable properties.

    activities: new Proxy({}, {
      get(target, p, receiver) {
        return (activities as any)[p];
      },
    }),

Reply:
We have this condition that prevents starting an activity worker:

```
 if (this.options.activities === undefined || Object.keys(this.options.activities).length === 0)
```

You can still use `Proxy` for dynamic activities if you proxy a non empty object:

```ts
activities: new Proxy(activities, {
  get(target, p) {
    const f = (activities as any)[p];
    if (f == null) { return default }
    return f;
  },
}),
```

Reply:
But your point holds for if the activities are all non enumerable.